presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -170,8 +182,12 @@ else if (element instanceof LikeClause) {
                     inheritedProperties = likeTableMetadata.getMetadata().getProperties();
                 }
 
-                likeTableMetadata.getColumns().stream()
-                        .filter(column -> !column.isHidden())
+                accessControl.filterColumns(
+                        session.toSecurityContext(),
+                        tableName.asCatalogSchemaTableName(),
+                        likeTableMetadata.getColumns().stream()
+                                .filter(column -> !column.isHidden())
+                                .collect(toUnmodifiableList()))
```

Comment:
`toImmutableList()` let's be consistent with the current convention. Consider changing the convention in the separate pull request.

Reply:
Wow. I haven't even noticed. I guess this is the default idea behaviour now.